core/src/main/java/org/apache/accumulo/core/lock/ServiceLock.java

```diff
@@ -558,6 +558,11 @@ public synchronized void unlock() throws InterruptedException, KeeperException {
     LOG.debug("[{}] Deleting all at path {} due to unlock", vmLockPrefix, pathToDelete);
     ZooUtil.recursiveDelete(zooKeeper, pathToDelete, NodeMissingPolicy.SKIP);
 
+    // Wait for the delete to happen on the server before exiting method
+    while (zooKeeper.exists(pathToDelete, null) != null) {
```

Comment:
Do we need a timeout here in case something goes wrong with the delete? We may not as any server issues get rethrown and should break but was something I was wondering, if there's a scenario we could get stuck in this loop forever.

Reply:
`ServiceLock.unlock`is only called from the server processes when they are shutting down and from the tests. Not waiting for the locks to be removed caused issues in tests where the cluster was restarted in between test methods. I think for the tests we need to wait, but maybe I'll add some logging here so that we know that the server is waiting to shutdown because the locks are not removed. The user can then have the option to forcefully terminate the process if they want.